chain/client/src/client.rs

```diff
@@ -2260,19 +2257,15 @@ impl Client {
                     }
                 }
             };
-            if let Some(err) = self
-                .runtime_adapter
-                .validate_tx(
-                    gas_price,
-                    Some(state_root),
-                    tx,
-                    false,
-                    &epoch_id,
-                    protocol_version,
-                    receiver_congestion_info,
-                )
-                .expect("no storage errors")
-            {
+            if let Err(err) = self.runtime_adapter.validate_tx(
```

Comment:
Oh man the old semantics of returning Some in the error case was mad. This is a great improvement!